src/v/utils/retry_chain_node.h

```diff
@@ -383,6 +389,10 @@ class retry_chain_logger final {
     }
 
 private:
+    void __attribute__((noinline)) do_log(
```

Comment:
I think noinline is unnecessary if the function body is in the .cc file

Reply:
I think that with LTO it is possible to inline a function defined in cpp
also, the attributes shows the intention clearly

Reply:
@ztlpn @Lazin nice catch / fix ya'll. that was a wild bug